diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.14.0/root/etc/e-smith/web/functions/backup mezzanine_patched_e-smith-backup-1.14.0/root/etc/e-smith/web/functions/backup
--- e-smith-backup-1.14.0/root/etc/e-smith/web/functions/backup	2008-03-24 16:03:01.000000000 -0600
+++ mezzanine_patched_e-smith-backup-1.14.0/root/etc/e-smith/web/functions/backup	2008-03-24 16:02:22.000000000 -0600
@@ -1264,7 +1264,7 @@
 	unless(system("/sbin/e-smith/signal-event", "post-upgrade") == 0)
 	{
 		$rec->set_prop('errmsg', $rec->prop('errmsg').
-			'<br>ERR_UPDATING_CONF_AFTER_TAPE_RESTORE');
+			'ERR_UPDATING_CONF_AFTER_TAPE_RESTORE');
 		$rec->delete_prop('state');
 	    die ($fm->localise('ERR_UPDATING_CONF_AFTER_TAPE_RESTORE'));
 	}
