--- e-smith-backup-2.0.0/root/etc/e-smith/web/functions/backup.password	2008-10-29 13:24:01.000000000 -0600
+++ e-smith-backup-2.0.0/root/etc/e-smith/web/functions/backup	2008-10-29 13:35:32.000000000 -0600
@@ -1089,7 +1089,7 @@
 	print $fm->localise('WORKSTN_BACKUP_SHARE'), ' ', $backupwkFolder, '<br/>';
 	if ($VFSType eq 'cifs') {
     	    print $fm->localise('LOGIN'), ' ', $backupwkLogin, '<br/>';
-    	    print $fm->localise('PASSWORD'), ' ', $backupwkPassword, '<br/>';
+    	    print $fm->localise('PASSWORD'), ' ********<br/>';
 	    }
 	print $fm->localise('WORKSTN_BACKUP_SETSNUM'), ' ', $setsNumber, '<br/>';
 	print $fm->localise('WORKSTN_BACKUP_DAYSINSET'), ' ', $filesinset, '<br/>';
@@ -1330,7 +1330,7 @@
 			    $fm->localise('PASSWORD')
 			),
 
-	    esmith::cgi::genCell ($q, $q->textfield (-name  => 'backupwkPassword',
+	    esmith::cgi::genCell ($q, $q->password_field (-name  => 'backupwkPassword',
 						-override => 1,
 						-default => $backupwkPassword,
 						-size => 20)),
