diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.15.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/backup mezzanine_patched_e-smith-backup-1.15.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/backup
--- e-smith-backup-1.15.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/backup	2008-08-04 07:45:33.000000000 -0600
+++ mezzanine_patched_e-smith-backup-1.15.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/backup	2008-08-04 07:45:17.000000000 -0600
@@ -158,10 +158,6 @@
    <trans>Select an action</trans>
   </entry>
   <entry>
-   <base>PERFORM</base>
-   <trans>Perform</trans>
-  </entry>
-  <entry>
    <base>X_BACKUP_OR_RESTORE</base>
    <trans>X Backup or restore server data</trans>
   </entry>
@@ -208,10 +204,6 @@
    <trans>Backup file to restore from</trans>
   </entry>
   <entry>
-   <base>RESTORE</base>
-   <trans>Restore</trans>
-  </entry>
-  <entry>
    <base>VERIFY_BACKUP_DESC</base>
    <trans>
     <![CDATA[
@@ -273,10 +265,6 @@
    <trans>Error occured while restoring initial group of 'www'.</trans>
   </entry>
   <entry>
-   <base>ERR_UPDATING_SYSCONF</base>
-   <trans>Error occurred while updating system configuration.</trans>
-  </entry>
-  <entry>
    <base>RESTORE_FAILED</base>
    <trans>Restore failed! The backup file was incomplete.</trans>
   </entry>
@@ -619,10 +607,6 @@
    </trans>
   </entry>
   <entry>
-   <base>WORKSTN_PASSWORD</base>
-   <trans>Password</trans>
-  </entry>
-  <entry>
    <base>WORKSTATION_BACKUP_DEST</base>
    <trans>Backup workstation settings</trans>
   </entry>
diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup mezzanine_patched_e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup
--- e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup	2008-08-04 07:45:33.000000000 -0600
+++ mezzanine_patched_e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup	2008-08-04 07:45:17.000000000 -0600
@@ -1326,7 +1326,7 @@
 	print $q->Tr(
 		    esmith::cgi::genCell(
 			    $q,
-			    $fm->localise('WORKSTN_PASSWORD')
+			    $fm->localise('PASSWORD')
 			),
 
 	    esmith::cgi::genCell ($q, $q->textfield (-name  => 'backupwkPassword',
@@ -2263,7 +2263,7 @@
 		system("/usr/sbin/usermod", "-g", "$www_gid", "www") == 0
 		    or warn ($fm->localise('ERR_RESTORING_INITIAL_GRP')."\n");
 		system("/sbin/e-smith/signal-event", "post-upgrade") == 0
-		    or die ($fm->localise('ERR_UPDATING_SYSCONF')."\n");
+		    or die ($fm->localise('ERROR_UPDATING_CONFIGURATION')."\n");
 	    }
 	    else
 	    {
