diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup mezzanine_patched_e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup
--- e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup	2008-08-07 13:47:56.000000000 -0600
+++ mezzanine_patched_e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup	2008-08-07 13:47:34.000000000 -0600
@@ -1077,7 +1077,8 @@
 
     if ($rec) {
 	print $fm->localise('WORKSTN_BACKUP_DESC');
-	print $fm->localise('WORKSTN_BACKUP_ENABLED'), $q->b(' '.$backupwk_status), '.<br/>';
+	print $fm->localise('WORKSTN_BACKUP_ENABLED'), 
+            $q->b(' '.$fm->localise(uc($backupwk_status))), '.<br/>';
         if ($VFSType eq 'usb') {
 	    print $fm->localise('WORKSTN_BACKUP_USB'), '<br/>';
 	    }
