diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.14.0/root/etc/e-smith/templates/etc/crontab/backup mezzanine_patched_e-smith-backup-1.14.0/root/etc/e-smith/templates/etc/crontab/backup
--- e-smith-backup-1.14.0/root/etc/e-smith/templates/etc/crontab/backup	2008-07-26 08:49:34.000000000 -0600
+++ mezzanine_patched_e-smith-backup-1.14.0/root/etc/e-smith/templates/etc/crontab/backup	2008-07-26 08:49:13.000000000 -0600
@@ -29,7 +29,7 @@
 
     $OUT .=
 	"$reminderMin $reminderHour * * * root"
-	. " /bin/mt -f $device status | grep ONLINE > /dev/null 2>&1 ||"
+	. " /bin/mt -f $device status 2>&1 | grep ONLINE > /dev/null ||"
 	. ' echo "' . gettext("Remember to load the backup tape!") . '"'
 	. ' | mail -s "' . gettext("Remember to load the backup tape!") . '"'
         . " $reminderEmail\n";
