diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup mezzanine_patched_e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup
--- e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup	2008-03-25 11:29:03.000000000 -0600
+++ mezzanine_patched_e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup	2008-03-25 11:28:30.000000000 -0600
@@ -129,7 +129,7 @@
 	{
 		if($rec)
 		{
-			$msg = $fm->localise($rec->prop('errmsg') || '');
+			$msg = $rec->prop('errmsg') || '';
 			$rec->delete_prop('errmsg');
 		}
 	}
@@ -975,13 +975,13 @@
 
 	unless(system("/sbin/e-smith/signal-event", "pre-restore") == 0)
 	{
-		$rec->set_prop('errmsg', 'ERR_PRE_RESTORE');
+		$rec->set_prop('errmsg', $fm->localize('ERR_PRE_RESTORE'));
 		$rec->delete_prop('state');
 	    die ($fm->localise('ERR_PRE_RESTORE'),"\n");
 	}
 	unless(system("/sbin/e-smith/signal-event", "restore-tape") == 0)
 	{
-		$rec->set_prop('errmsg', 'ERR_RESTORING_FROM_TAPE');
+		$rec->set_prop('errmsg', $fm->localize('ERR_RESTORING_FROM_TAPE'));
 		$rec->delete_prop('state');
 	    die ($fm->localise('ERR_RESTORING_FROM_TAPE')."\n");
 	}
@@ -992,19 +992,20 @@
 
 	unless(system("/usr/sbin/groupmod", "-g", "$www_gid", "www") == 0)
 	{
-		$rec->set_prop('errmsg', $rec->prop('errmsg').'<br>ERR_RESTORING_GID');
+		$rec->set_prop('errmsg', $rec->prop('errmsg').'<br>'.
+		    $fm->localize('ERR_RESTORING_GID'));
 	    warn ($fm->localise('ERR_RESTORING_GID')."\n");
 	}
 	unless(system("/usr/sbin/usermod", "-g", "$www_gid", "www") == 0)
 	{
-		$rec->set_prop('errmsg', $rec->prop('errmsg').
-			'<br>ERR_RESTORING_INITIAL_GRP');
+		$rec->set_prop('errmsg', $rec->prop('errmsg').'<br>'.
+			$fm->localize('ERR_RESTORING_INITIAL_GRP'));
 	    warn ($fm->localise('ERR_RESTORING_INITIAL_GRP')."\n");
 	}
 	unless(system("/sbin/e-smith/signal-event", "post-upgrade") == 0)
 	{
-		$rec->set_prop('errmsg', $rec->prop('errmsg').
-			'ERR_UPDATING_CONF_AFTER_TAPE_RESTORE');
+		$rec->set_prop('errmsg', $rec->prop('errmsg').'<br>'.
+			$fm->localize('ERR_UPDATING_CONF_AFTER_TAPE_RESTORE'));
 		$rec->delete_prop('state');
 	    die ($fm->localise('ERR_UPDATING_CONF_AFTER_TAPE_RESTORE'));
 	}
@@ -1026,7 +1027,7 @@
 	# Error
 
 	$rec->delete_prop('state');
-	$rec->set_prop('errmsg', 'COULD_NOT_FORK');
+	$rec->set_prop('errmsg', $fm->localize('COULD_NOT_FORK'));
 	die ($fm->localise("COULD_NOT_FORK")."$!\n");
     }
 }
