diff -Naur e-smith-backup-2.2.0-orig/root/usr/lib/perl5/site_perl/esmith/Backup.pm e-smith-backup-2.2.0/root/usr/lib/perl5/site_perl/esmith/Backup.pm
--- e-smith-backup-2.2.0-orig/root/usr/lib/perl5/site_perl/esmith/Backup.pm	2010-05-13 17:59:09.551254429 +0200
+++ e-smith-backup-2.2.0/root/usr/lib/perl5/site_perl/esmith/Backup.pm	2010-05-13 18:00:14.940021174 +0200
@@ -81,7 +81,7 @@
         'etc/shadow',
         'etc/group',
         'etc/gshadow',
-        'var/lib/samba/private/secrets.tdb',
+        'etc/samba/secrets.tdb',
         'etc/samba/smbpasswd',
         'etc/smbpasswd',
     );
