--- e-smith-backup-2.0.0/root/etc/e-smith/templates/etc/dar/DailyBackup.dcf/60exclude-compression.nocompressogg	2008-10-08 04:34:30.000000000 +1100
+++ e-smith-backup-2.0.0/root/etc/e-smith/templates/etc/dar/DailyBackup.dcf/60exclude-compression	2008-10-21 21:54:58.000000000 +1100
@@ -25,5 +25,10 @@
 --exclude-compression "*.rpm"
 --exclude-compression "*.iso"
 --exclude-compression "*.bz2"
+--exclude-compression "*.flac"
+--exclude-compression "*.ogv"
+--exclude-compression "*.oga"
+--exclude-compression "*.ogx"
+--exclude-compression "*.ogg"
 '
 }

