diff -up e-smith-backup-2.0.0/root/etc/e-smith/web/functions/backup.smbpassword2smbpasswd e-smith-backup-2.2.0/root/etc/e-smith/web/functions/backup
--- e-smith-backup-2.0.0/root/etc/e-smith/web/functions/backup.smbpassword2smbpasswd	2010-06-14 08:49:05.000000000 +0200
+++ e-smith-backup-2.0.0/root/etc/e-smith/web/functions/backup	2010-06-14 08:50:58.000000000 +0200
@@ -2425,7 +2425,7 @@ sub performWorkstnRestore
 	while (<RD>)
 	{
 	    $complete++ if /etc\/smbpasswd$/;
-	    $complete++ if /etc\/samba\/smbpassword$/;
+	    $complete++ if /etc\/samba\/smbpasswd$/;
 	    print "<li>$_</li>\n";
 	}
 
