diff -up e-smith-backup-2.2.0/root/usr/lib/perl5/site_perl/esmith/Backup.pm.adjust-for-samba-tdb-locations e-smith-backup-2.2.0/root/usr/lib/perl5/site_perl/esmith/Backup.pm
--- e-smith-backup-2.2.0/root/usr/lib/perl5/site_perl/esmith/Backup.pm.adjust-for-samba-tdb-locations	2007-09-05 21:45:52.000000000 +0200
+++ e-smith-backup-2.2.0/root/usr/lib/perl5/site_perl/esmith/Backup.pm	2010-03-17 19:04:01.000000000 +0100
@@ -81,7 +81,7 @@ sub restore_list
         'etc/shadow',
         'etc/group',
         'etc/gshadow',
-        'etc/samba/secrets.tdb',
+        'var/lib/samba/private/secrets.tdb',
         'etc/samba/smbpasswd',
         'etc/smbpasswd',
     );
 
