diff -up e-smith-backup-2.0.0/createlinks.CIFScredentials2 e-smith-backup-2.0.0/createlinks
--- e-smith-backup-2.0.0/createlinks.CIFScredentials2	2009-08-26 07:23:54.000000000 +0200
+++ e-smith-backup-2.0.0/createlinks	2009-08-26 07:26:16.000000000 +0200
@@ -18,12 +18,13 @@ my $panel = "manager";
 panel_link("backup", $panel);
 
 #--------------------------------------------------
-# functions for bootstrap-console-save
+# events for bootstrap-console-save
 #--------------------------------------------------
 
 $event = "bootstrap-console-save";
 
 event_link("reset-restore-idle-flag", $event, "95");
+templates2events("/etc/dar/CIFScredentials", $event);
 
 #--------------------------------------------------
 # events for conf-backup action
