diff -ruN e-smith-backup-2.2.0.old/root/etc/e-smith/events/actions/workstation-backup-dar e-smith-backup-2.2.0/root/etc/e-smith/events/actions/workstation-backup-dar
--- e-smith-backup-2.2.0.old/root/etc/e-smith/events/actions/workstation-backup-dar	2010-05-06 08:53:54.000000000 +0100
+++ e-smith-backup-2.2.0/root/etc/e-smith/events/actions/workstation-backup-dar	2010-05-06 08:54:54.000000000 +0100
@@ -79,6 +79,7 @@
 my $tmpdir = $mntdir . "/tmp_dir";
 
 $report .= "From: admin-backup\n";
+$report .= "To: admin\n";
 $report .= "Subject: Daily Backup Report\n\n";
 $report .= "================================== \n";
 $report .= "DAILY BACKUP TO WORKSTATION REPORT \n";
