diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.14.0/root/etc/e-smith/web/functions/backup mezzanine_patched_e-smith-backup-1.14.0/root/etc/e-smith/web/functions/backup
--- e-smith-backup-1.14.0/root/etc/e-smith/web/functions/backup	2007-07-01 08:19:51.000000000 -0600
+++ mezzanine_patched_e-smith-backup-1.14.0/root/etc/e-smith/web/functions/backup	2007-07-01 08:12:38.000000000 -0600
@@ -51,6 +51,7 @@
 my $es_backup = new esmith::Backup or die "Couldn't create Backup object\n";
 
 my @directories = $es_backup->restore_list;
+@directories = grep { -e "/$_" } @directories;
 
 # Unbuffer standard output so that files and directories are listed as
 # they are restored
