diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.14.0/root/etc/e-smith/events/actions/eject-tape mezzanine_patched_e-smith-backup-1.14.0/root/etc/e-smith/events/actions/eject-tape
--- e-smith-backup-1.14.0/root/etc/e-smith/events/actions/eject-tape	2007-01-05 09:15:08.000000000 -0700
+++ mezzanine_patched_e-smith-backup-1.14.0/root/etc/e-smith/events/actions/eject-tape	2007-01-05 09:10:34.000000000 -0700
@@ -7,7 +7,7 @@
 
 my $event = shift;
 my $cause = shift;
-exit 0 if not defined $cause or $cause eq "desktop";
+exit 0 if not defined $cause or $cause ne "tape";
 
 my $confdb = esmith::ConfigDB->open_ro;
 my $backup = $confdb->get('backup') or die "No backup db entry found\n";
diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.14.0/root/etc/e-smith/events/actions/rewind-tape mezzanine_patched_e-smith-backup-1.14.0/root/etc/e-smith/events/actions/rewind-tape
--- e-smith-backup-1.14.0/root/etc/e-smith/events/actions/rewind-tape	2004-03-01 13:23:34.000000000 -0700
+++ mezzanine_patched_e-smith-backup-1.14.0/root/etc/e-smith/events/actions/rewind-tape	2007-01-05 09:11:22.000000000 -0700
@@ -28,7 +28,7 @@
 
 my $event = shift;
 my $cause = shift;
-exit 0 if not defined $cause or $cause eq "desktop";
+exit 0 if not defined $cause or $cause ne "tape";
 
 my $confdb = esmith::ConfigDB->open_ro;
 my $backup = $confdb->get('backup') or die "No backup db entry found\n";
