--- e-smith-backup-1.13.4/root/etc/e-smith/web/functions/backup.orig	2006-02-16 18:47:17.000000000 -0500
+++ e-smith-backup-1.13.4/root/etc/e-smith/web/functions/backup	2006-02-16 18:51:15.000000000 -0500
@@ -809,9 +809,12 @@
 
     ($backupHour, $backupMin) = split (":", $backupTime, -1);
 
-    if ($backupHour > 12)
+    if ($backupHour > 11)
     {
-	$backupHour -= 12;
+	if ($backupHour > 12)
+	{
+	    $backupHour -= 12;
+	}
 	$backupAMPM = 'PM';
     }
 
