diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.13.4/createlinks mezzanine_patched_e-smith-backup-1.13.4/createlinks
--- e-smith-backup-1.13.4/createlinks	2006-02-01 17:07:12.000000000 -0500
+++ mezzanine_patched_e-smith-backup-1.13.4/createlinks	2006-02-01 10:47:24.000000000 -0500
@@ -53,3 +53,8 @@
 {
     event_link("merge-system-files", $event, "00");
 }
+
+foreach $event (qw(restore-tape))
+{
+    event_link("restore-from-disk", $event, "40");
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.13.4/root/etc/e-smith/events/actions/restore-from-disk mezzanine_patched_e-smith-backup-1.13.4/root/etc/e-smith/events/actions/restore-from-disk
--- e-smith-backup-1.13.4/root/etc/e-smith/events/actions/restore-from-disk	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-backup-1.13.4/root/etc/e-smith/events/actions/restore-from-disk	2006-01-31 15:43:45.000000000 -0500
@@ -0,0 +1,56 @@
+#!/usr/bin/perl
+
+#----------------------------------------------------------------------
+# copyright (C) 2006 Charlie Brady
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+#
+#----------------------------------------------------------------------
+
+package esmith;
+
+use warnings;
+use strict;
+use esmith::ConfigDB;
+use esmith::Backup;
+
+my $event = shift;
+my $path = shift;
+
+my $confdb = esmith::ConfigDB->open_ro;
+my $backup = $confdb->get('backup') or die "No backup db entry found\n";
+my $program = $backup->prop('Program');
+
+exit 0 unless $program eq 'CopyFromDisk';
+
+die("Root path to restore from must be provided\n") unless defined $path;
+chdir $path or die "Could not chdir to $path: $!";
+
+my $backup_db = new esmith::Backup or die "Couldn't create Backup object\n";
+
+my @restore = $backup_db->restore_list;
+
+die "Can't fork: $!" unless defined(my $pid = open(STDIN, "-|"));
+if ($pid)
+{
+    # parent
+    chdir "/";
+    exec("cpio", "-duvmp", "/tmp/test.restore");
+}
+else
+{
+    exec("find", @restore);
+}
+
