diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.15.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/backup mezzanine_patched_e-smith-backup-1.15.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/backup
--- e-smith-backup-1.15.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/backup	2008-03-08 19:18:52.000000000 +1100
+++ mezzanine_patched_e-smith-backup-1.15.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/backup	2008-03-08 19:13:54.000000000 +1100
@@ -7,6 +7,34 @@
    <base>BACKUP_DESC</base>
    <trans>
     <![CDATA[
+     <P>The server provides two ways to back up and restore
+     your server: using your local desktop or a tape drive.</P>
+
+     <P>The first method creates a copy of your server configuration and user
+     data files, and downloads it to your local desktop via your web browser.
+     Currently your configuration and data files total approximately
+     <b>$tarsize</b>. The backup file will be somewhat less than this,
+     depending on how compressible the data are. The \"Verify desktop backup
+     file\" option can be used to check the integrity of a desktop backup
+     file.</P>
+
+     <P>The tape backup method uses a software package called <I>$module</I>
+     to back up your entire hard disk to tape every night. This requires a
+     supported tape drive and a tape that is not write-protected. The backup
+     is performed automatically at the selected time every night (with a
+     reminder automatically e-mailed to the administrator during the day).
+     Currently your hard disk contains <b>$dumpsize</b> of data.</P>
+
+     <P>Both restore methods allow you to restore your configuration and user
+     data files. <b>Ideally, the restore should be performed on a freshly
+     installed server</b>.</P>
+     ]]>
+    </trans>
+  </entry>
+  <entry>
+   <base>BACKUP_DESC_DAR</base>
+   <trans>
+    <![CDATA[
      <P>The server provides three ways to back up and restore
      your server: using your local desktop, a tape drive or a 
      workstation on your LAN.</P>
diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup mezzanine_patched_e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup
--- e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup	2008-03-08 19:18:52.000000000 +1100
+++ mezzanine_patched_e-smith-backup-1.15.0/root/etc/e-smith/web/functions/backup	2008-03-08 19:15:58.000000000 +1100
@@ -173,7 +173,7 @@
     }
 
     # Hack to evaluate embedded vars
-    print eval 'return "'.$fm->localise('BACKUP_DESC').'\n";';
+    print eval 'return "'.$fm->localise('BACKUP_DESC_DAR').'\n";';
 
     if ($tarsize =~ /Tb/ or $tarsize =~ /(\d+)Gb/ and $1 >= 2)
     {
