diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.13.4/root/etc/e-smith/web/functions/backup mezzanine_patched_e-smith-backup-1.13.4/root/etc/e-smith/web/functions/backup
--- e-smith-backup-1.13.4/root/etc/e-smith/web/functions/backup	2006-02-06 10:12:24.000000000 -0500
+++ mezzanine_patched_e-smith-backup-1.13.4/root/etc/e-smith/web/functions/backup	2006-02-06 10:11:55.000000000 -0500
@@ -809,9 +809,9 @@
 
     ($backupHour, $backupMin) = split (":", $backupTime, -1);
 
-    if ($backupHour > 12)
+    if ($backupHour >= 12)
     {
-	$backupHour -= 12;
+	if ($backupHour > 12) { $backupHour -= 12; }
 	$backupAMPM = 'PM';
     }
 
