diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.14.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/backup mezzanine_patched_e-smith-backup-1.14.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/backup
--- e-smith-backup-1.14.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/backup	2003-06-28 07:32:13.000000000 +1000
+++ mezzanine_patched_e-smith-backup-1.14.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/backup	2006-03-15 10:53:26.325157754 +1100
@@ -448,4 +448,15 @@
 	<trans>Backup or restore</trans>
     </entry>
 
+  <entry>
+	<base>BACKUP_DESKTOP_TOO_BIG</base>
+	<trans>
+	<![CDATA[
+	<div class="error">
+	Your server has too much data for a reliable backup to desktop.
+	</div>
+	]]>
+	</trans>
+  </entry>
+
  </lexicon>
diff -Nur -x '*.orig' -x '*.rej' e-smith-backup-1.14.0/root/etc/e-smith/web/functions/backup mezzanine_patched_e-smith-backup-1.14.0/root/etc/e-smith/web/functions/backup
--- e-smith-backup-1.14.0/root/etc/e-smith/web/functions/backup	2006-03-15 04:08:45.000000000 +1100
+++ mezzanine_patched_e-smith-backup-1.14.0/root/etc/e-smith/web/functions/backup	2006-03-15 10:52:22.265260893 +1100
@@ -155,6 +155,11 @@
     # Hack to evaluate embedded vars
     print eval 'return "'.$fm->localise('BACKUP_DESC').'\n";';
 
+    if ($dumpsize =~ /Tb/ or $dumpsize =~ /(\d+)Gb/ and $1 >= 2)
+    {
+        print $fm->localise("BACKUP_DESKTOP_TOO_BIG"), "\n"
+    }
+
     print $q->h2($fm->localise("BACKUP_CONFIG_STATUS")),"\n";
 
     my $backup_status = $conf->get('backup');
